spec/content/automate/ManageIQ/Cloud/VM/Provisioning/StateMachines/VMProvision_VM.class/__methods__/retry_state_spec.rb

```diff
@@ -0,0 +1,72 @@
+require_domain_file
+
+describe ManageIQ::Automate::Service::Generic::StateMachines::VMProvision_VM::RetryState do
+  describe "#retry_state" do
+    let(:user) { FactoryGirl.create(:user) }
+    let(:ems) { FactoryGirl.create(:ems_openstack_with_authentication) }
+    let(:vm_template) { FactoryGirl.create(:template_openstack, :ext_management_system => ems) }
+    let(:miq_request) do
+      FactoryGirl.create(:miq_provision_request,
+                         :provision_type => 'template',
+                         :state          => 'pending',
+                         :status         => 'Ok',
+                         :src_vm_id      => vm_template.id,
+                         :requester      => user)
+    end
+    let(:miq_server) { MiqServer.new }
+    let(:provision) do
+      MiqProvision.new.tap do |prov|
+        prov.miq_request = miq_request
+        prov.message = "An error occurred while provisioning Instance 1"
+      end
+    end
+    let(:root_object) do
+      Spec::Support::MiqAeMockObject.new.tap do |ro|
+        ro["miq_provision"] = provision
+        ro["miq_server"]    = miq_server
+        ro["ae_result"]     = "error"
+        ro["ae_next_state"] = "some_state"
+      end
+    end
+    let(:ae_service) { Spec::Support::MiqAeMockService.new(root_object) }
+
+    it "retries and increments count because of a matching error" do
+      allow(ae_service).to receive(:inputs).and_return({})
```

Comment:
@mansam @gmcculloug 
If this PR https://github.com/ManageIQ/manageiq/pull/13912 gets merged then you dont need to have this line.
You can set input to be a hash from the spec if you need it.